File: SwipeBack/app/src/main/java/github/why168/swipeback/activity/MainActivity.java
package github.why168.swipeback.activity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import github.why168.swipeback.R;
import github.why168.swipeback.base.BaseActivity;

/**
 * MainActivity
 *
 * @author Edwin.Wu
 * @version 2017/2/7 17:30
 * @since JDK1.8
 */
public class MainActivity extends BaseActivity implements View.OnClickListener {
    private Button button_edit_text_activity;
    private Button button_view_pager_activity;

    /**
     * 重写isSupportSwipeBack是否开启滑动返回，默认开启
     *
     * @return false关闭，true支持
     */
    @Override
    public boolean isSupportSwipeBack() {
        return false;
    }

    @Override
    protected int getLayoutResId() {
        return R.layout.activity_main;
    }

    @Override
    protected void setUpView() {
        button_edit_text_activity = (Button) findViewById(R.id.button_edit_text_activity);
        button_view_pager_activity = (Button) findViewById(R.id.button_view_pager_activity);
    }

    @Override
    protected void setUpData(Bundle savedInstanceState) {
        button_edit_text_activity.setOnClickListener(this);
        button_view_pager_activity.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button_edit_text_activity:
                mSwipeBackHelper.forward(EditTextActivity.class);
                break;
            case R.id.button_view_pager_activity:
                mSwipeBackHelper.forward(ViewPagerActivity.class);
                break;
            default:
                break;
        }
    }

    @Override
    public void onBackPressed() {
        finish();
    }
}

File: KotlinLearning/app/src/main/java/com/github/why168/kotlinlearn/MainActivity.kt
package com.github.why168.kotlinlearn

import android.annotation.SuppressLint
import android.os.Build
import android.os.Bundle
import android.view.View
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Toast
import com.github.why168.kotlinlearn.api.APIWrapper
import com.github.why168.kotlinlearn.api.SchedulerUtils
import com.github.why168.kotlinlearn.base.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {
    override fun getLayoutResId(): Int {
        return R.layout.activity_main
    }

    @SuppressLint("SetJavaScriptEnabled", "SetTextI18n")
    override fun setUpData(savedInstanceState: Bundle?) {
        // setText
        main_txt.text = "KotlinLearn"
        main_txt.setOnClickListener({
            Toast.makeText(it.context, "KotlinLearn", Toast.LENGTH_SHORT).show()
        })

        main_httpButton.setOnClickListener {
            Toast.makeText(this, "网络请求", Toast.LENGTH_LONG).show()
            // HTTP
            APIWrapper
                    .getFirstHomeData(1)
                    .compose(SchedulerUtils.ioToMain())
                    .subscribe({ homeBean ->
                        Toast.makeText(this, "请求成功", Toast.LENGTH_LONG).show()
                        println(homeBean.toString())
                    }, { error ->
                        Toast.makeText(this, "请求失败", Toast.LENGTH_LONG).show()
                        error.printStackTrace()
                    })
        }


        // WebView
        val settings = main_WebView.settings
        settings.javaScriptEnabled = true
        main_WebView.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                    view!!.loadUrl(request!!.url.toString())
                }
                return true
            }
        }
        main_WebView.loadUrl("https://github.com/why168/JavaProjects/blob/master/Kotlinlang/README.md")
    }

    fun onClickTest(view: View) {
        Toast.makeText(this, "测试onClick", Toast.LENGTH_LONG).show()
    }
}


